protocols/manage/propagate.go

```diff
@@ -0,0 +1,199 @@
+package manage
+
+import (
+	"sync"
+
+	"time"
+
+	"reflect"
+
+	"github.com/dedis/cothority/lib/dbg"
+	"github.com/dedis/cothority/lib/network"
+	"github.com/dedis/cothority/lib/sda"
+)
+
+func init() {
+	sda.ProtocolRegisterName("Propagate", NewPropagateProtocol)
+}
+
+// Propagate is a protocol that sends some data to all attached nodes
+// and waits for confirmation before returning.
+type Propagate struct {
+	*sda.TreeNodeInstance
+	onData    func(network.ProtocolMessage)
+	onDoneCb  func(int)
+	sd        *PropagateSendData
+	ChannelSD chan struct {
+		*sda.TreeNode
+		PropagateSendData
+	}
+	ChannelReply chan struct {
+		*sda.TreeNode
+		PropagateReply
+	}
+
+	received int
+	subtree  int
+	sync.Mutex
+}
+
+// CreateProtocolEntity is the necessary interface to start a protocol.
+// It is implemented by Service and Overlay.
+type CreateProtocolEntity interface {
+	CreateProtocol(t *sda.Tree, name string) (sda.ProtocolInstance, error)
+	Entity() *network.Entity
+}
+
+// PropagateSendData is the message to pass the data to the children
+type PropagateSendData struct {
+	// Data is the data to transmit
+	Data []byte
+	// Msec is the timeout in milliseconds
+	Msec int
+}
+
+// PropagateReply is sent from the children back to the root
+type PropagateReply struct {
+	Level int
+}
+
+// PropagateStartAndWaitSDA starts the propagation protocol and blocks till everything
+// is OK or the timeout has been reached.
+// This is used when you want to start a protocol without a service.
+func PropagateStartAndWaitSDA(o *sda.Overlay, el *sda.EntityList, msg network.ProtocolMessage, msec int, f func(network.ProtocolMessage)) (int, error) {
```

Comment:
Further question: why do we allow such propage protocol using *only* SDA ? The whole point of the propagate protocol is to broadcast a data through the tree right ? What's the use of using only sda if  the data will be lost after ? lost means no `Service` will use it somehow ? I'm sure you have a use case, what is it :D ?